### PR TITLE
Replace changelog/updating with release notes and towncrier

### PR DESCRIPTION
**Pull request by jedcunningham**
This proposes a few changes, primarily combining our existing changelog and updating docs into a single "release notes" doc, and start using towncrier to help generate it.

More details and context on the mailing list: https://lists.apache.org/thread/1bch0s4tr4w3vvxwt12z3vx0cyh0qp4o

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -49,7 +49,7 @@
 !.flake8
 !.dockerignore
 !pytest.ini
-!CHANGELOG.txt
+!RELEASE_NOTES.txt
```
Comment (uranusjr) [inline .dockerignore]: This should be `RELEASE_NOTES.rst`?

Comment (jedcunningham) [reply .dockerignore]: Why, yes, yes it should!

Comment (jedcunningham) [reply .dockerignore]:
```suggestion
!RELEASE_NOTES.rst
```

**Review by potiuk — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -266,7 +266,7 @@ repos:
           The word(s) should be in lowercase." && exec codespell "$@"' --
         language: python
         types: [text]
-        exclude: ^airflow/_vendor/|^CHANGELOG\.txt$|^airflow/www/static/css/material-icons\.css$|^images/.*$
+        exclude: ^airflow/_vendor/|^RELEASE_NOTES\.txt$|^airflow/www/static/css/material-icons\.css$|^images/.*$
```
Comment (blag) [inline .pre-commit-config.yaml:269]:
```suggestion
        exclude: ^airflow/_vendor/|^RELEASE_NOTES\.rst$|^airflow/www/static/css/material-icons\.css$|^images/.*$
```

**Review by blag — CHANGES_REQUESTED** [review]
One typo to fix.

Also: should the `RELEASE_NOTES.rst` file be included in the `.rat-excludes` file?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jedcunningham) [reply .pre-commit-config.yaml:269]: Ugh! strikes again! I think that's the last one thankfully.

**Comment by jedcunningham** [comment]
> Also: should the `RELEASE_NOTES.rst` file be included in the `.rat-excludes` file?

No, it doesn't need to be as it has the license in it.

**Review by ashb — APPROVED** [review]

**Pull request merged**